### DISCUSSION TITLE
[Q&A] How to add TAG to SNMP devices discovered&inventoried by gpli-agent?

### DISCUSSION THREAD

**Discussion by gallogabor** [glpi-project/glpi-agent#324] [Q&A]
### Documentation & bug reporting acknowledgment

Yes, I read it

### Describe your problem

I'm a newbie to GLPI, started with a fresh 10.0.6 in a Docker container and a Linux desktop with the agent version 1.4-1.
(The server is running in a hosting environment remotely, with a Let's Encrypt certificate, through a reverse proxy - I wish this is the way it intended to be run.)

After some try-and-error (the documentation not so clear for me, at least at first), I have to activate features many places and reinstall the agent with all functionality instead of typical.
Finally I managed to do a network discovery and inventory of SNMP devices on my own home network.

Some of them showed up in the asset lists (a printer and a NAS).

My problem is that these SNMP devices all added to the root entity. No TAG attached to them, so the rule I created to assign devices to entities by tag do nothing for these devices. The computer running the agent also added to inventory, and using the tag I specified in the config it's associated automatically with the entity I set in the rule.

I'd like to use GLPI to manage multiple (remote) client's inventories, and I'd like to use TAG to identify which asset comes from which client's network (I didn't found a way yet, to assing entities to devices by agent or anything similar, only by tag). But as I know now, only computers associated with specified client by tag, other devices aren't.

Am I do something wrong or this functionality missing from the agent as of now?

**Comment by g-bougard** [comment]
Hi @gallogabor 
can you still share your assign to entity rule definition which works for computers ?

**Reply by gallogabor** [reply]
Hi @g-bougard 

Of course, here it is:

Criteria
- Source - is - GLPI Inventory
- Inventory tag - is - GG
Actions
- Entity - Assign - Root>GG

I tried to test the agent with computer and SNMP inventory to find out what information it sends to GLPI. Unfortunately I don't know how to do it. If I run CLI commands "glpi-inventory" or "glpi-netinverory", they not acting exactly as I run tasks managed by GLPI server, with the agent.

But one difference is easily visible: "glpi-inventory" command have "--tag" option while "glpi-netinventory" do not have this option. So I can't run manual inventory with TAG specified and added for SNMP devices. Maybe remote controlled agent run also can't add TAG to this type of inventory yet.

One thing I found while examining GLPI is that I can set different "Agent base URL" for every entity. May I aks that if I set different URLs here (I think these are virtual URLs), and point the respective agent to this entity-specific URL, every item inventoried by the agent will be added to that entity regardless of TAG and/or TAG rules?

Thank you for your time!

P.s.: I dont want to mix multiple things in one thread, but I would like to avoid opening multiple questions for a small misunderstanding. So, may I ask that if I participate in translation to Hungarian, which version to choose? 10.0 ot 10.1? I downloaded 10.1 PO file from Trasifex, translated and corrected many strings and uploaded back to 10.1. Of course I test it on a 10.0.6, not using nightlies or similar. Is this the proper way or stick to 10.0 translation and re-translate/copy string for 10.1 when it will be available? May I upload this 10.1 PO file to 10.0, too?

**Reply by g-bougard** [reply]
You should probably not use "GLPI Inventory" as source as this is indeed GLPI Native inventory which handles inventories. In that case, you can use "GLPI" as source.

With GLPI Native inventory, the tag is indeed associated to the agent. We should expect the tag is set from the glpi-agent service expected to run a netdiscovery/netinventory task.
Actually, tag is not supported in netdiscovery & netinventory XML format.

"Agent base URL" is here for another purpose and this is not a virtual address. This URL will be used for deploy, collect & ESX tasks by GlpiInventory plugin and will be transmitted to the agent running these tasks. You should better leave empty by default.

About your translation question, this is definitively not the right place for it as this is really specific to GLPI project. You should better ask on the forum. As far I know, the team uses [transifex](https://explore.transifex.com/glpi/GLPI/). I see there hungarian is at 92% and you just probably have to review the current state and help to update it.
